packages/select-v2/src/useAllowCreate.ts

```diff
@@ -0,0 +1,90 @@
+import { computed, ref } from 'vue'
+import type { ISelectProps } from './token'
+import type { Option } from './select.types'
+
+export function useAllowCreate(props: ISelectProps, states) {
+
+  const createOptionCount = ref(0)
+  const cachedSelectedOption = ref<Option>(null)
+
+  const enableAllowCreateMode = computed(() => {
+    return props.allowCreate && props.filterable
+  })
+
+  function hasExistingOption(query: string) {
+    if (!props.options) {
+      return false
+    }
+    return props.options.concat(states.createdOptions).some(option => {
+      return option.value === query
+    })
```

Comment:
Can we test them individually instead of allocating new array and perform a `some` on it?

Reply:
E.g.
```ts
hasValue = o => option.value === query

return props.options.some(hasValue) || states.createdOptions.some(hasValue)
```